src/GraphQL/Types/Composite/AutoRegisteringObjectGraphType.cs

```diff
@@ -73,7 +74,85 @@ protected virtual IEnumerable<FieldType> ProvideFields()
+                return (null, MemberResolver.Create(methodInfo));
+            }
+
+            List<LambdaExpression> expressions = new();
+            QueryArguments queryArguments = new();
+            foreach (var parameterInfo in methodInfo.GetParameters())
+            {
+                var argumentInfo = (ArgumentInformation?)_getArgumentInformationInternalMethodInfo
+                    .MakeGenericMethod(parameterInfo.ParameterType)
+                    .Invoke(this, new object[] { fieldType, parameterInfo })!;
+                var (queryArgument, expression) = argumentInfo.ConstructQueryArgument();
+                if (queryArgument != null)
+                {
+                    ApplyArgumentAttributes(parameterInfo, queryArgument);
+                    queryArguments.Add(queryArgument);
+                }
+                expression ??= AutoRegisteringHelper.GetParameterExpression(
+                    parameterInfo.ParameterType,
+                    queryArgument ?? throw new InvalidOperationException("Invalid response from ConstructQueryArgument: queryArgument and expression cannot both be null"));
+                expressions.Add(expression);
+            }
+            var resolver = new MethodResolver(methodInfo, expressions);
+            return (queryArguments, resolver);
+        }
+
+        private static readonly MethodInfo _getArgumentInformationInternalMethodInfo = typeof(AutoRegisteringObjectGraphType<TSourceType>).GetMethod(nameof(GetArgumentInformationInternal), BindingFlags.NonPublic | BindingFlags.Instance);
+        private ArgumentInformation GetArgumentInformationInternal<TReturnType>(FieldType fieldType, ParameterInfo parameterInfo)
+            => GetArgumentInformation<TReturnType>(fieldType, parameterInfo);
+
+        /// <summary>
+        /// Applies <see cref="GraphQLAttribute"/> attributes defined on the supplied <see cref="ParameterInfo"/>
+        /// to the specified <see cref="QueryArgument"/>.
+        /// </summary>
+        protected virtual void ApplyArgumentAttributes(ParameterInfo parameterInfo, QueryArgument queryArgument)
+        {
+            // Apply derivatives of GraphQLAttribute
+            var attributes = parameterInfo.GetCustomAttributes<GraphQLAttribute>();
+            foreach (var attr in attributes)
+            {
+                attr.Modify(queryArgument);
+            }
+        }
+
+        /// <summary>
+        /// Analyzes a method parameter and returns an instance of <see cref="ArgumentInformation"/>
+        /// containing information necessary to build a <see cref="QueryArgument"/> and <see cref="IFieldResolver"/>.
+        /// Also applies any <see cref="GraphQLAttribute"/> attributes defined on the <see cref="ParameterInfo"/>
+        /// to the returned <see cref="ArgumentInformation"/> instance.
+        /// </summary>
+        protected virtual ArgumentInformation GetArgumentInformation<TReturnType>(FieldType fieldType, ParameterInfo parameterInfo)
```

Comment:
So `TReturnType` always equals to `ParameterInfo.ParameterType` and we provide both.

Reply:
I had renamed all the `TReturnType` to `TParameterType` but must have missed this one.  I'll rename this one also for clarity.

Reply:
> So `TReturnType` always equals to `ParameterInfo.ParameterType` and we provide both.

Correct